index.ts

```diff
@@ -36,6 +59,76 @@ app.post('/wh/updateDocsJson', async (req, res) => {
 	})
 })
 
+app.post('/tutorial/feedback', feedbackRateLimit, async (req, res) => {
+	const body = JSON.parse(JSON.stringify(req.body)) as FeedbackRequestBody
+	// validation of request body
+	if (
+		typeof body.turnstileToken !== "string" ||
+		// type of string should be "up" or "down" but it's checked later
+		typeof body.feedback !== "string" ||
+		typeof body.route !== "string"
+  	)
+    return res
+		.status(400)
+		.send({
+			successful: false,
+			error: "You have something missing in your request!",
+		});
+	if (body.feedback !== "up" && body.feedback !== "down")
+		return res
+			.status(400)
+			.send({
+				successful: false,
+				error: "Feedback must be either 'up' or 'down'!",
+			});
```

Comment:
Why check like that? Just check up & firstly

Reply:
the request body can be tampered by the user, so we check the types to not get any javascript errors that could crash the bot